pkg/virtctl/objectgraph/objectgraph_test.go

```diff
@@ -0,0 +1,149 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package objectgraph_test
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/mock/gomock"
+
+	k8smetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/pointer"
+	"kubevirt.io/kubevirt/pkg/virtctl/testing"
+)
+
+var _ = Describe("Object graph command", func() {
+	const (
+		objectGraphCommand = "objectgraph"
+		vmName             = "testvm"
+	)
+
+	var (
+		ctrl         *gomock.Controller
+		vmInterface  *kubecli.MockVirtualMachineInterface
+		vmiInterface *kubecli.MockVirtualMachineInstanceInterface
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		kubecli.GetKubevirtClientFromClientConfig = kubecli.GetMockKubevirtClientFromClientConfig
+		kubecli.MockKubevirtClientInstance = kubecli.NewMockKubevirtClient(ctrl)
+		vmiInterface = kubecli.NewMockVirtualMachineInstanceInterface(ctrl)
+		vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
+	})
+
+	Context("For VM", func() {
```

Comment:
Last question, can't you make some of the cases tables so it tests the scenarios for VM and VMI? Label selectors and non existing resource also should work as expected with VMIs?

Reply:
As long as the functionality is covered I'm fine with that

Reply:
I would like to see the cases with label selectors and a non existing resource also being covered for VMIs. It's a different call to the cluster API afterall.